agent/lib/kontena/network_adapters/weave_executor.rb

```diff
@@ -0,0 +1,94 @@
+require_relative '../logging'
+require_relative '../helpers/image_helper'
+require_relative '../helpers/iface_helper'
+
+module Kontena::NetworkAdapters
+  class WeaveExecutor
+    include Celluloid
+    include Celluloid::Notifications
+    include Kontena::Logging
+    include Kontena::Helpers::ImageHelper
+
+    def initialize(autostart = true)
+      @images_exist = false
+      #$CELLULOID_DEBUG = true
+      info 'initialized'
+      async.ensure_images if autostart
+    end
+
+    # @param [Array<String>] cmd
+    def execute(cmd)
+      sleep 0.5 until @images_exist
```

Comment:
Are these still guaranteed to execute consecutively and in order?

Particularly for the startup sequence, the `launch-router` must happen before the `expose` etc.